### PR TITLE
[WIP] Make dark theme selectable in preferences

### PR DESCRIPTION
**Pull request by ngrant123**
----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
darkthemeprefbox
darkthemeoption

This pull request allows the user to pick the dark theme option in the preferences box. After the user restarts the application the UI would have the dark theme in it. I also added the dark theme in the preferences box. An critique on my coding will be greatly appreciated.

### CODE REVIEW

File: src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java [hunk 1]
```diff
@@ -10,58 +12,151 @@
...
+
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            accessDarkTheme.setSelected(true);
+        }
+
+    }
+
+    private void copy(File file1,File file2) throws IOException {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java]: You can use the copy method from FileUtil

File: src/main/java/org/jabref/gui/FXDialogService.java [hunk 2]
```diff
@@ -64,6 +64,7 @@ private static FXDialog createDialog(AlertType type, String title, String conten
         alert.setHeaderText(null);
         alert.setContentText(content);
         alert.getDialogPane().setMinHeight(Region.USE_PREF_SIZE);
+        alert.showAndWait();
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/FXDialogService.java]: This is not a good idea to add it here as is the internal method for creating a dialog. Normally you call one of the public method from the Dialog Service, get a dialog object back and then call show and wait on that object. Just look in other places how it's done

File: src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java [hunk 3]
```diff
@@ -10,58 +12,151 @@
...
         return container;
     }
 
+    //Basically replace each of the files with each other
+    //Ultimately you have to read the contents of the file and replace it
+    private void addLogic(CheckBox accessDarkTheme) throws URISyntaxException {
+
+        if(accessDarkTheme.isSelected()==true){
+
+            File basecaseFile= new File(DEFAULT_PATH_MAIN_CSS);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java]: Please use the nio Path methods here. Have a look at the exporter tests to see how to access a resource file

File: src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java [hunk 4]
```diff
@@ -10,58 +12,151 @@
...
+            File darkthemecssFile= new File(DARK_THEME_PATH_CSS);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/AppearancePrefsTab.java]: I don't understand why the copy and replace is necessary. I thought it's just pointing the JavaFX loader to the new path 🤔

File: src/main/java/org/jabref/gui/util/ThemeLoader.java [hunk 5]
```diff
@@ -51,6 +55,9 @@ public ThemeLoader(FileUpdateMonitor fileUpdateMonitor) {
      */
     public void installBaseCss(Scene scene, JabRefPreferences preferences) {
         addAndWatchForChanges(scene, DEFAULT_PATH_MAIN_CSS, 0);
+        addAndWatchForChanges(scene,DARK_THEME_PATH_CSS,0);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/util/ThemeLoader.java]: It should work without copying (and modifying) the css files if you add here something like:
````
if (preferences.useDarkTheme) {
   addAndWatchForChanges(scene,DARK_THEME_PATH_CSS, 1);
}
````

**Comment by stefan-kolb** [comment]
I would rather habe a dropdown to select the two different styles than a checkbox for the dark theme.

**Comment by ngrant123** [comment]
Cool I'll look at everyone's advice and suggestions and try to fix it.

File: src/main/java/org/jabref/gui/util/ThemeLoader.java (hunk 5 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/util/ThemeLoader.java]: Do we really need to watch for changes? Maybe just have a second method which just adds/loads the css and doesn't  watch for changes.

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/util/ThemeLoader.java]: It only watches for changes if the corresponding command line flag is passed, i.e. only for us developers.

**Comment by tobiasdiez** [comment]
https://github.com/JabRef/jabref/pull/4372 was a bit quicker to implement this. Sorry!

**Pull request closed without merging**